linux/hid.c

```diff
@@ -963,6 +963,12 @@ int HID_API_EXPORT hid_write(hid_device *dev, const unsigned char *data, size_t
 {
 	int bytes_written;
 
+	if (!data || (length == 0)) {
+		errno = EINVAL;
```

Comment:
I guess hidraw is using system calls a lot and if any of those fails, errno will be set, so having it here, even if "randomly" should be fine

Reply:
I was trying to "blend in" with the rest of the source using `register_device_error` and this looked out of place:

```c
        register_device_error(dev, strerror(EINVAL));
```

I'm ok with whatever is most in the style of the source, not trying cause more work or make waves :)

Reply:
Thanks for your help

Reply:
Glad to help, I've been a consumer of this project for some time and I'm happy to contribute back to it in some small way.